### PR TITLE
solution v0.1

### PR DESCRIPTION
**Pull request by nzrii-kysel**
[DEMO LINK](https://nzrii-kysel.github.io/react_people-table-advanced/)

### CODE REVIEW

File: src/components/SortLink/SortLink.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  return (
+    <Link
+      to={{
+        search: getSearchWithSort(sort),
+      }}
+    >
+      <span className="icon">
+        <i
+          className={classNames(
```
Comment (anna-inozemets) [inline src/components/SortLink/SortLink.tsx:45]: Pay attention for this classNames. For the first click we should see fa-sort-up, not fa-sort-down, you can see at the demo link

File: src/utils/getFilteredPeople.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          return person.sex === 'm';
+        case 'f':
+          return person.sex === 'f';
+        default:
+          return true;
+      }
+    });
+  }
+
+  if (centuries.length) {
```
Comment (anna-inozemets) [inline src/utils/getFilteredPeople.ts:35]: I guess, in this case we should take just born cuntury into account, you can see the difference between that filter and filter on demo link

**Review by anna-inozemets — APPROVED** [review]
Good and clear code!)

File: src/components/SortLink/SortLink.tsx (hunk 1 above)
Comment (nzrii-kysel) [reply src/components/SortLink/SortLink.tsx:45]: i was intentional, I believe the way I did it makes more sense

File: src/utils/getFilteredPeople.ts (hunk 2 above)
Comment (nzrii-kysel) [reply src/utils/getFilteredPeople.ts:35]: you are right, thanks

File: src/App.tsx [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
-      <div className="section">
-        <div className="container">
-          <h1 className="title">Home Page</h1>
-          <h1 className="title">Page not found</h1>
-          <PeoplePage />
-        </div>
+    <div className="section">
+      <div className="container">
+        <Routes>
+          <Route path="/" element={<h1 className="title">Home Page</h1>} />
```
Comment (Anna-Tkachenko) [inline src/App.tsx:14]: let's create separate component for the element

File: src/App.tsx [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
+          <Route path="/home" element={<Navigate to="/" replace />} />
+
+          <Route path="/people">
+            <Route index element={<PeoplePage />} />
+            <Route path=":person" element={<PeoplePage />} />
+          </Route>
+
+          <Route path="*" element={<h1 className="title">Page not found</h1>} />
```
Comment (Anna-Tkachenko) [inline src/App.tsx:22]: same here

File: src/components/Navbar/NavbarLink/NavbarLink.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  children?: React.ReactNode,
+};
+
+export const NavbarLink: React.FC<Props> = React.memo(({
+  pathname,
+  preserveParams = false,
+  children,
+}) => {
+  const location = useLocation();
+  const search = preserveParams ? location.search : '';
```
Comment (Anna-Tkachenko) [inline src/components/Navbar/NavbarLink/NavbarLink.tsx:17]: 🔥

**Review by Anna-Tkachenko — APPROVED** [review]
Well Done! ⭐️
Clean and great solution!

**Pull request closed without merging**